### ISSUE TITLE
Remaining issues on Fedora 33/34

### ISSUE THREAD

**Issue by hroncok** [serge-sans-paille/pythran#1639]
Hello @serge-sans-paille o/

I've tried to build recent master branch as a Fedora package (with Python 3.9.0rc1).

https://koji.fedoraproject.org/koji/taskinfo?taskID=51460855

Here are the remaining test failures:

There are a lot of:

```
E   ImportError: /builddir/build/BUILD/pythran-master/test_matmul_operator.cpython-39-x86_64-linux-gnu.so: undefined symbol: cblas_cgemv
```

Not sure if that is related to FlexiBLAS / OpenBLAS or Python 3.9.

I've also tried to build with FlexiBLAS, but there were compilation failures in tests :/

https://koji.fedoraproject.org/koji/taskinfo?taskID=51471040

I've also tried to build it on Travis CI:

https://github.com/hroncok/pythran/pull/1

There is juts one failure:

```
____________________ TestNumpyRandom.test_numpy_logistic0b _____________________
self = <pythran.tests.test_numpy_random.TestNumpyRandom testMethod=test_numpy_logistic0b>
    def test_numpy_logistic0b(self):
        """ Check logistic with 2 argument with mean and variance. """
        code = """
        def numpy_logistic0b(size):
            from numpy.random import logistic
            from numpy import var, mean, pi
            u = 2.
            s = 2
            rmean = u
            rvar = ((s**2*pi**2)/3)
            a = logistic(u, s, size)
            return (abs(mean(a) - rmean) < 0.05 and abs(var(a) - rvar) < .05)
        """
>       self.run_test(code, 10 ** 6, numpy_logistic0b=[int])
pythran/tests/test_numpy_random.py:1399: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
pythran/tests/__init__.py:318: in run_test
    self.assertAlmostEqual(python_ref, pythran_res)
pythran/tests/__init__.py:144: in assertAlmostEqual
    self.assertEqual(ref, res)
E   AssertionError: True != False
----------------------------- Captured stdout call -----------------------------
Python result:  True
Pythran result:  False
Type of Pythran res :  <class 'bool'>
Type of Python ref :  <class 'numpy.bool_'>
```

Travis CI has Python 3.9.0rc1+ from git.

Any help is appreciated.

**Comment by hroncok** [comment]
OK, I've been able to solve:

```
E   ImportError: /builddir/build/BUILD/pythran-master/test_matmul_operator.cpython-39-x86_64-linux-gnu.so: undefined symbol: cblas_cgemv
```

With:

```
sed -i 's|libs=|libs=openblas|' pythran/pythran-linux*.cfg
```

Running a new build.

**Comment by serge-sans-paille** [comment]
And the other faiure on travis is just a defect we have when testing random generators, we try to estimate the mean and variance, but well, that's random numbers so sometimes the test fail.

**Comment by serge-sans-paille** [comment]
btw I plan to do a release in the forthcoming days,

**Comment by hroncok** [comment]
> btw I plan to do a release in the forthcoming days,

I just want to be ready and verify it works for Fedora before you ship it.

**Comment by hroncok** [comment]
x86_64 succeeded :balloon:

**Comment by hroncok** [comment]
i686:

```
________________________ TestCases.test__histogram_run0 ________________________
[gw3] linux -- Python 3.9.0 /usr/bin/python3
self = <pythran.tests.TestFromDir.TestFunctor object at 0xea3e9e20>
    def __call__(self):
        if "unittest.skip" in self.module_code:
            return self.test_env.skipTest("Marked as skippable")
    
        if "unittest.python3.skip" in self.module_code:
            return self.test_env.skipTest("Marked as skippable")
    
        # resolve import locally to where the tests are located
        sys.path.insert(0, self.test_env.path)
    
>       self.test_env.run_test_case(self.module_code, self.module_name,
                                    self.runas, module_dir=self.module_dir,
                                    **self.specs)
pythran/tests/__init__.py:454: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
pythran/tests/__init__.py:262: in run_test_case
    self.assertAlmostEqual(python_ref, pythran_res)
pythran/tests/__init__.py:119: in assertAlmostEqual
    self.assertAlmostEqual(ref_v, res_v)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
self = <pythran.tests.test_cases.TestCases testMethod=test__histogram_run0>
ref = array([1, 1, 2]), res = array([0, 0, 0], dtype=int32)
    def assertAlmostEqual(self, ref, res):
        """
        Improved version of assertAlmostEqual.
    
        This new version supports nan, complex and ndarray.
        """
        self.check_type(ref, res)
        if isinstance(res, (list, tuple)):
            self.assertEqual(len(res), len(ref))
            for res_v, ref_v in zip(res, ref):
                self.assertAlmostEqual(ref_v, res_v)
        elif isinstance(ref, ndarray):
>           npt.assert_array_almost_equal(ref, res)
E           AssertionError: 
E           Arrays are not almost equal to 6 decimals
E           
E           Mismatched elements: 3 / 3 (100%)
E           Max absolute difference: 2
E           Max relative difference: inf
E            x: array([1, 1, 2])
E            y: array([0, 0, 0], dtype=int32)
pythran/tests/__init__.py:121: AssertionError
------------------------------ Captured log setup ------------------------------
WARNING  root:spec.py:94 No pythran specification, nothing will be exported
----------------------------- Captured stdout call -----------------------------
Python result:  (array([1, 1, 2]), array([0, 1, 2, 3]))
Pythran result:  (array([0, 0, 0], dtype=int32), array([0, 1, 2, 3]))
Type of Pythran res :  <class 'tuple'>
Type of Python ref :  <class 'tuple'>
Type of Pythran res :  <class 'numpy.ndarray'>
Type of Python ref :  <class 'numpy.ndarray'>
_________________________ TestNumpyFunc0.test_tofile2 __________________________
[gw5] linux -- Python 3.9.0 /usr/bin/python3
self = <pythran.tests.test_numpy_func0.TestNumpyFunc0 testMethod=test_tofile2>
    def test_tofile2(self):
        temp_name = tempfile.mkstemp()[1]
>       x = numpy.random.randint(0,2**32,1000).astype(numpy.uint32)
pythran/tests/test_numpy_func0.py:423: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
mtrand.pyx:744: in numpy.random.mtrand.RandomState.randint
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
>   ???
E   ValueError: high is out of bounds for int32
_bounded_integers.pyx:1343: ValueError
________________________ TestNumpyFunc0.test_fromfile2 _________________________
[gw0] linux -- Python 3.9.0 /usr/bin/python3
self = <pythran.tests.test_numpy_func0.TestNumpyFunc0 testMethod=test_fromfile2>
    def test_fromfile2(self):
        temp_name = tempfile.mkstemp()[1]
>       x = numpy.random.randint(0,2**32,1000).astype(numpy.uint32)
pythran/tests/test_numpy_func0.py:465: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
mtrand.pyx:744: in numpy.random.mtrand.RandomState.randint
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
>   ???
E   ValueError: high is out of bounds for int32
_bounded_integers.pyx:1343: ValueError
_________________________ TestNumpyFunc2.test_asarray5 _________________________
[gw1] linux -- Python 3.9.0 /usr/bin/python3
self = <pythran.tests.test_numpy_func2.TestNumpyFunc2 testMethod=test_asarray5>
    def test_asarray5(self):
>       self.run_test("def np_asarray5(a):\n from numpy import asarray\n return asarray(a)", 1, np_asarray5=[int])
pythran/tests/test_numpy_func2.py:494: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
pythran/tests/__init__.py:318: in run_test
    self.assertAlmostEqual(python_ref, pythran_res)
pythran/tests/__init__.py:115: in assertAlmostEqual
    self.check_type(ref, res)
pythran/tests/__init__.py:91: in check_type
    self.check_type(ref.item(0), res)
pythran/tests/__init__.py:107: in check_type
    self.assertIsInstance(res, type(ref))
E   AssertionError: 1 is not an instance of <class 'int'>
----------------------------- Captured stdout call -----------------------------
Python result:  1
Pythran result:  1
Type of Pythran res :  <class 'numpy.intc'>
Type of Python ref :  <class 'numpy.ndarray'>
Type of Pythran res :  <class 'numpy.intc'>
Type of Python ref :  <class 'int'>
```

**Comment by serge-sans-paille** [comment]
Closed by #1640
